components/application-connectivity-validator/internal/validationproxy/handler.go

```diff
@@ -164,9 +172,21 @@ func (ph *proxyHandler) getClientIDsFromResource(applicationName string) ([]stri
 
 func (ph *proxyHandler) mapRequestToProxy(path string) (*httputil.ReverseProxy, apperrors.AppError) {
 	switch {
-	case strings.HasPrefix(path, ph.eventServicePathPrefixV1), strings.HasPrefix(path, ph.eventServicePathPrefixV2):
+	// For a cluster which is BEB enabled, events reaching with prefix /{application}/v1/events will be routed to /{application}/v1/events endpoint of event-publisher-proxy
+	// For a cluster which is not BEB enabled, events reaching with prefix /{application}/events will be routed to /{application}/v1/events endpoint of event-service
+	case strings.HasPrefix(path, ph.eventServicePathPrefixV1):
+		return ph.eventsProxy, nil
+
+	// For a cluster which is BEB enabled, events reaching /{application}/v2/events will be routed to /publish endpoint of event-publisher-proxy
+	// For a cluster which is not BEB enabled, events reaching /{application}/v2/events will be routed to /{application}/v2/events endpoint of event-service
+	case strings.HasPrefix(path, ph.eventServicePathPrefixV2):
+		if ph.isBEBEnabled {
+			return ph.eventMeshProxy, nil
+		}
 		return ph.eventsProxy, nil
```

Comment:
Isn't that equivalent to `/{application}/v1/events` which contradicts the comment or does the endpoint depend on eventService version for an application?

Reply:
In case of `/v2/events` (which is deprecated), we handle it differently than `/v1/events`. Both types are handled by the same service, but two different endpoints.